svm/src/account_loader.rs

```diff
@@ -97,23 +100,34 @@ pub struct FeesOnlyTransaction {
 
 #[cfg_attr(feature = "dev-context-only-utils", derive(Clone))]
 pub(crate) struct AccountLoader<'a, CB: TransactionProcessingCallback> {
-    account_overrides: Option<&'a AccountOverrides>,
     pub(crate) program_cache: ProgramCacheForTxBatch,
     program_accounts: HashMap<Pubkey, (&'a Pubkey, u64)>,
+    account_cache: AHashMap<Pubkey, AccountSharedData>,
     callbacks: &'a CB,
     pub(crate) feature_set: Arc<FeatureSet>,
 }
 impl<'a, CB: TransactionProcessingCallback> AccountLoader<'a, CB> {
-    pub fn new(
+    pub fn new_with_account_cache_capacity(
         account_overrides: Option<&'a AccountOverrides>,
         program_cache: ProgramCacheForTxBatch,
         program_accounts: HashMap<Pubkey, (&'a Pubkey, u64)>,
         callbacks: &'a CB,
         feature_set: Arc<FeatureSet>,
+        capacity: usize,
     ) -> AccountLoader<'a, CB> {
+        let mut account_cache = AHashMap::with_capacity(capacity);
+
+        // SlotHistory may be overridden for simulation.
+        // No other uses of AccountOverrides are expected.
+        if let Some(slot_history) =
+            account_overrides.and_then(|overrides| overrides.get(&slot_history::id()))
+        {
+            account_cache.insert(slot_history::id(), slot_history.clone());
```

Comment:
Should we add a debug assert that the size of `account_overrides` is equal to `1` if sysvar is present and `0` otherwise? Or is it basically impossible now that we decreased the vis of the account overrides `set_account` function?

Reply:
it isnt possible, the only exposed setter is `set_slot_history` now (which accepts `AccountSharedData` and hardcodes the `Pubkey` as `sysvar::slot_history::id()`. we can remove `AccountOverrides` entirely now as a cleanup. im thinking ill look into it when i start on simd186 since id like to change something about simulation results, and this is related to simulation only